dev/tests/integration/testsuite/Magento/ProductAlert/_files/customer_unsubscribe_stock_rollback.php

```diff
@@ -0,0 +1,20 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+use Magento\Catalog\Api\ProductRepositoryInterface;
+use Magento\ProductAlert\Model\ResourceModel\Stock;
+use Magento\TestFramework\Helper\Bootstrap;
+
+$objectManager = Bootstrap::getObjectManager();
+$resource = $objectManager->get(Stock::class);
+
+$productRepository = $objectManager->create(ProductRepositoryInterface::class);
+$productId = $productRepository->get('simple-out-of-stock')->getId();
+
+$resource->getConnection()->delete(
+    $resource->getMainTable(),
+    'product_id = ' . $productId
```

Comment:
```
$resource->getConnection()->delete(
    $resource->getMainTable(),
    ['product_id = ?' => $productId]
)
```

Your code is absolutely safe, no security threats here, but using string concatenation in queries is bad behavior from the security perspective.

Reply:
Hello, thanks for pointing that out. Was changed.